commons/zenoh-buffers/src/zbuf.rs

```diff
@@ -658,7 +657,7 @@ impl Writer for ZBufWriter<'_> {
         let s = crate::unsafe_slice_mut!(cache.spare_capacity_mut(), ..len);
         // SAFETY: converting MaybeUninit<u8> into [u8] is safe because we are going to write on it.
```

Comment:
This is backwards. We should write and to it and _then_ `MaybeUninit::assume_init`.

Reply:
Tackling the uninitialized memory issue not the subject of this PR.